edb/schema/referencing.py

```diff
@@ -643,13 +757,21 @@ def _validate(self, schema, context):
                     context=self.source_context,
                 )
 
-    def _propagate_ref_op(self, schema, context, scls, cb):
-
+    def _propagate_ref_op(self,
+                          schema: s_schema.Schema,
+                          context: sd.CommandContext,
+                          scls: ReferencedInheritingObject,
+                          cb: Callable[[sd.Command, str], None]
+                          ) -> s_schema.Schema:
+        from edb.schema import pointers as s_pointers
         rec = context.current().enable_recursion
         context.current().enable_recursion = False
-
         referrer_ctx = self.get_referrer_context(context)
-        referrer = referrer_ctx.scls
+
+        # type ignore below, because mypy doesn't trust referrer_ctx to be an
+        # instance of sd.ObjectCommandContext (considers it <nothing>)
+        assert isinstance(referrer_ctx, sd.ObjectCommandContext)
+        referrer = referrer_ctx.scls  # type: ignore
```

Comment:
In this case, and another identical one, I used "type: ignore" because, while asserting that referrer_ctx is an instance of sd.ObjectCommandContext (or sd.ObjectCommandContext[so.Object]) should be sufficient, mypy still complains that "<nothing> as no property `scls`".

sd.ObjectCommandContext defines a property scls as generic "Object_T" which is bound to so.Object.

To me, this looks like a limitation in mypy, it looks like it gets confused for the generic class.

Reply:
Yes, that's fine.  The comment explaining why `type: ignore` is needed is sufficient.